<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to singleCellTK • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to singleCellTK">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_QC-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">3: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">4: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wevanjohnson/singleCellTK" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to singleCellTK</h1>
                        <h4 data-toc-skip class="author">David Jenkins</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA;Program in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:dfj@bu.edu" class="email">dfj@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Mohammed Muzamil Khan</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA;Program in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:mmkhan@bu.edu" class="email">mmkhan@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">W. Evan Johnson</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA;Program in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:wej@bu.edu" class="email">wej@bu.edu</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/wevanjohnson/singleCellTK/blob/HEAD/vignettes/v01-Introduction_to_singleCellTK.Rmd" class="external-link"><code>vignettes/v01-Introduction_to_singleCellTK.Rmd</code></a></small>
      <div class="hidden name"><code>v01-Introduction_to_singleCellTK.Rmd</code></div>

    </div>

    
    
<p>The Single Cell Toolkit (SCTK) is an interactive scRNA-Seq analysis package that allows a user to upload raw scRNA-Seq count matrices and perform downstream scRNA-Seq analysis interactively through a web interface, or through a set of R functions through the command line. The package is written in R with a graphical user interface (GUI) written in Shiny. Users can perform analysis with modules for filtering raw results, clustering, batch correction, differential expression, pathway enrichment, and scRNA-Seq study design, all in a simple to use point and click interface. The toolkit also supports command line data processing, and results can be loaded into the GUI for additional exploration and downstream analysis.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can download the release version of the Single Cell Toolkit <!-- in
[Bioconductor v3.10](https://bioconductor.org/packages/release/bioc/html/singleCellTK.html):

```r
if (!requireNamespace("BiocManager", quietly=TRUE))
  install.packages("BiocManager")
BiocManager::install("singleCellTK")
```

### Devel Version

You can download the development version of the Single Cell Toolkit in
[Bioconductor v3.11](https://bioconductor.org/packages/devel/bioc/html/singleCellTK.html)
or --> from this repository:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"wevanjohnson/singleCellTK"</span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="troubleshooting-installation">Troubleshooting Installation<a class="anchor" aria-label="anchor" href="#troubleshooting-installation"></a>
</h3>
<p>For the majority of users, the commands above will install the latest version of the singleCellTK without any errors. Rarely, you may encounter an error due to previously installed versions of some packages or missing packages that are required for the singleCellTK. If you encounter an error during installation, use the commands below to check the version of Bioconductor that is installed:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/version.html" class="external-link">version</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>If the version number is not 3.14 or higher, you must upgrade Bioconductor to install the toolkit:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>After you install Bioconductor 3.14 or higher, you should be able to install the toolkit using <code>devtools::install_github("wevanjohnson/singleCellTK")</code>. If you still encounter an error, ensure your Bioconductor packages are up to date by running the following command.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>If the command above does not return <code>TRUE</code>, run the following command to update your R packages:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Then, try to install the toolkit again:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"wevanjohnson/singleCellTK"</span><span class="op">)</span></code></pre></div>
<p>If you still encounter an error, please <a href="mailto:dfj@bu.edu">contact us</a> and we’d be happy to help.</p>
</div>
</div>
<div class="section level2">
<h2 id="data-structure">Data Structure<a class="anchor" aria-label="anchor" href="#data-structure"></a>
</h2>
<p>The Single Cell Toolkit uses a <a href="https://www.bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a> object to store data matrices along with annotation information, metadata, and reduced dimensionality data (PCA, t-SNE, etc.).</p>
</div>
<div class="section level2">
<h2 id="run-the-shiny-app">Run the Shiny App<a class="anchor" aria-label="anchor" href="#run-the-shiny-app"></a>
</h2>
<div class="section level3">
<h3 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p>Example data is available within the app. To get started, simply run the singleCellTK function:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://compbiomed.github.io/sctk_docs/" class="external-link">singleCellTK</a></span><span class="op">)</span>
<span class="fu"><a href="../reference/singleCellTK.html">singleCellTK</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="upload-data-directly-through-the-shiny-app">Upload data directly through the shiny app<a class="anchor" aria-label="anchor" href="#upload-data-directly-through-the-shiny-app"></a>
</h3>
<p>To upload count matrices and annotation information stored as text files, run the singleCellTK function:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://compbiomed.github.io/sctk_docs/" class="external-link">singleCellTK</a></span><span class="op">)</span>
<span class="fu"><a href="../reference/singleCellTK.html">singleCellTK</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Then, follow data upload instructions in the “Upload Tab” vignette.</p>
</div>
<div class="section level3">
<h3 id="load-data-from-a-singlecellexperiment-object">Load data from a SingleCellExperiment object<a class="anchor" aria-label="anchor" href="#load-data-from-a-singlecellexperiment-object"></a>
</h3>
<div class="section level4">
<h4 id="creating-a-singlecellexperiment-object">Creating a SingleCellExperiment object<a class="anchor" aria-label="anchor" href="#creating-a-singlecellexperiment-object"></a>
</h4>
<p>To create a SingleCellExperiment object, we have provided the <code><a href="../reference/createSCE.html">createSCE()</a></code> function:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://compbiomed.github.io/sctk_docs/" class="external-link">singleCellTK</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: package 'S4Vectors' was built under R version 4.1.3</span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"mouseBrainSubsetSCE"</span><span class="op">)</span>
<span class="va">counts_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">mouseBrainSubsetSCE</span>, <span class="st">"counts"</span><span class="op">)</span>
<span class="va">sample_annot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">mouseBrainSubsetSCE</span><span class="op">)</span>
<span class="va">row_annot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">mouseBrainSubsetSCE</span><span class="op">)</span>
<span class="va">newSCE</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createSCE.html">createSCE</a></span><span class="op">(</span>assayFile <span class="op">=</span> <span class="va">counts_mat</span>, annotFile <span class="op">=</span> <span class="va">sample_annot</span>, 
                    featureFile <span class="op">=</span> <span class="va">row_annot</span>, assayName <span class="op">=</span> <span class="st">"counts"</span>,
                    inputDataFrames <span class="op">=</span> <span class="cn">TRUE</span>, createLogCounts <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="loading-data-stored-in-a-singlecellexperiment-object">Loading data stored in a SingleCellExperiment object<a class="anchor" aria-label="anchor" href="#loading-data-stored-in-a-singlecellexperiment-object"></a>
</h4>
<p>Once a SingleCellExperiment object is created, the object can be loaded directly from the R console:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/singleCellTK.html">singleCellTK</a></span><span class="op">(</span><span class="va">newSCE</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="vignettes">Vignettes<a class="anchor" aria-label="anchor" href="#vignettes"></a>
</h2>
<p>To help you get started with the SCTK, we have prepared several vignettes in two categories: interactive analysis and R console analysis.</p>
<div class="section level3">
<h3 id="interactive-analysis">Interactive Analysis<a class="anchor" aria-label="anchor" href="#interactive-analysis"></a>
</h3>
<ul>
<li><a href="https://compbiomed.github.io/sctk_docs/articles/v03-tab01_Upload.html" class="external-link">Upload Tab</a></li>
<li><a href="https://compbiomed.github.io/sctk_docs/articles/v04-tab02_Data-Summary-and-Filtering.html" class="external-link">Data Summary and Filtering Tab</a></li>
<li><a href="https://compbiomed.github.io/sctk_docs/articles/v05-tab03_Visualization-and-Clustering.html" class="external-link">Visualization and Clustering Tab</a></li>
<li><a href="https://compbiomed.github.io/sctk_docs/articles/v06-tab04_Batch-Correction.html" class="external-link">Batch Correction Tab</a></li>
<li><a href="https://compbiomed.github.io/sctk_docs/articles/v07-tab05_Differential-Expression.html" class="external-link">Differential Expression Tab</a></li>
<li><a href="https://compbiomed.github.io/sctk_docs/articles/v08-tab06_Pathway-Activity-Analysis.html" class="external-link">Pathway Activity Analysis Tab</a></li>
<li><a href="https://compbiomed.github.io/sctk_docs/articles/v09-tab07_Sample-Size.html" class="external-link">Sample Size Tab</a></li>
</ul>
</div>
<div class="section level3">
<h3 id="r-console-analysis">R Console Analysis<a class="anchor" aria-label="anchor" href="#r-console-analysis"></a>
</h3>
<ul>
<li><a href="v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data in the Single Cell Toolkit</a></li>
<li><a href="https://compbiomed.github.io/sctk_docs/articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html" class="external-link">Aligning and Quantifying scRNA-Seq Data</a></li>
</ul>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] singleCellTK_1.9.1          DelayedArray_0.20.0        </span>
<span class="co">##  [3] Matrix_1.4-0                SingleCellExperiment_1.16.0</span>
<span class="co">##  [5] SummarizedExperiment_1.24.0 Biobase_2.54.0             </span>
<span class="co">##  [7] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        </span>
<span class="co">##  [9] IRanges_2.28.0              S4Vectors_0.32.4           </span>
<span class="co">## [11] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       </span>
<span class="co">## [13] matrixStats_0.62.0          BiocStyle_2.22.0           </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] XVector_0.34.0         bslib_0.3.1            compiler_4.1.2        </span>
<span class="co">##  [4] BiocManager_1.30.16    jquerylib_0.1.4        zlibbioc_1.40.0       </span>
<span class="co">##  [7] bitops_1.0-7           tools_4.1.2            digest_0.6.29         </span>
<span class="co">## [10] lattice_0.20-45        jsonlite_1.8.0         evaluate_0.15         </span>
<span class="co">## [13] memoise_2.0.1          rlang_1.0.2            cli_3.2.0             </span>
<span class="co">## [16] rstudioapi_0.13        yaml_2.3.5             pkgdown_2.0.2         </span>
<span class="co">## [19] xfun_0.30              fastmap_1.1.0          GenomeInfoDbData_1.2.7</span>
<span class="co">## [22] stringr_1.4.0          knitr_1.38             desc_1.4.1            </span>
<span class="co">## [25] fs_1.5.2               sass_0.4.1             systemfonts_1.0.4     </span>
<span class="co">## [28] grid_4.1.2             rprojroot_2.0.3        R6_2.5.1              </span>
<span class="co">## [31] textshaping_0.3.6      rmarkdown_2.14         bookdown_0.25         </span>
<span class="co">## [34] GSVAdata_1.30.0        purrr_0.3.4            magrittr_2.0.3        </span>
<span class="co">## [37] htmltools_0.5.2        ragg_1.2.2             stringi_1.7.6         </span>
<span class="co">## [40] RCurl_1.98-1.6         cachem_1.0.6</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
