<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>QC and Filtering Tab • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="QC and Filtering Tab">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_QC-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">3: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">4: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wevanjohnson/singleCellTK" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>QC and Filtering Tab</h1>
            <h3 data-toc-skip class="subtitle">Single Cell Toolkit</h3>
                        <h4 data-toc-skip class="author">David Jenkins</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA;Program in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:dfj@bu.edu" class="email">dfj@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Mohammed Muzamil Khan</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA;Program in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:mmkhan@bu.edu" class="email">mmkhan@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">W. Evan Johnson</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA;Program in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:wej@bu.edu" class="email">wej@bu.edu</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/wevanjohnson/singleCellTK/blob/HEAD/vignettes/v04-tab02_QC-and-Filtering.Rmd" class="external-link"><code>vignettes/v04-tab02_QC-and-Filtering.Rmd</code></a></small>
      <div class="hidden name"><code>v04-tab02_QC-and-Filtering.Rmd</code></div>

    </div>

    
    
<p><img src="img/summary_01.png"></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The quality control (QC) and filtering tab provides several summary statistics and ways to filter your data. The summary statistics and filtering are performed on the selected assay, which can be changed using the “Select Assay” drop-down.</p>
</div>
<div class="section level2">
<h2 id="data-summary-table">Data summary table<a class="anchor" aria-label="anchor" href="#data-summary-table"></a>
</h2>
<p><img src="img/summary_02.png" width="400"></p>
<p>The summary table provides several summary statistics about your data including:</p>
<ul>
<li>The number of cells</li>
<li>The number of genes</li>
<li>The average number of reads per cell</li>
<li>The average number of genes per cell</li>
<li>Cells with fewer than the selected cutoff expressed genes</li>
<li>Genes with no expression across all cells</li>
</ul>
</div>
<div class="section level2">
<h2 id="filtering">Filtering<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h2>
<p>Several filtering options are available</p>
<div class="section level3">
<h3 id="delete-outlier-genes-and-cells">Delete outlier genes and cells<a class="anchor" aria-label="anchor" href="#delete-outlier-genes-and-cells"></a>
</h3>
<p><img src="img/summary_03.png" width="400"></p>
<p>Data can be filtered by user selected cutoff values including:</p>
<ul>
<li>Remove genes with no expression across all cells. These genes are uninformative</li>
<li>Minimum detected genes per cell. Cells with few expressed genes could indicate an empty well.</li>
<li>Percent low gene expression. This filter will remove the genes with the least expression in the dataset</li>
<li>If there are specific cells that should be excluded from the dataset</li>
</ul>
<p>When you have set the cutoff filters to the desired level, you can filter data with the “Filter Data” button.</p>
</div>
<div class="section level3">
<h3 id="reset-to-orginal-data">Reset to orginal data<a class="anchor" aria-label="anchor" href="#reset-to-orginal-data"></a>
</h3>
<p>The SCTK saves a copy of the originally uploaded data, which can be restored by clicking the “Reset All” button in the “Delete Outliers” section.</p>
</div>
<div class="section level3">
<h3 id="filter-cells-by-annotation">Filter cells by annotation<a class="anchor" aria-label="anchor" href="#filter-cells-by-annotation"></a>
</h3>
<p><img src="img/summary_04.png" width="400"></p>
<p>Data stored in the annotation data frame can be used for sample filtering. Choose a column from the annotation data frame, the values from this column to keep, and click “Filter.”</p>
</div>
<div class="section level3">
<h3 id="filter-genes-by-feature-annotation">Filter genes by feature annotation<a class="anchor" aria-label="anchor" href="#filter-genes-by-feature-annotation"></a>
</h3>
<p><img src="img/summary_05.png" width="400"></p>
<p>Similarly, gene annotations can be used to filter genes. Select a column from the row data frame and the values to keep and click “Filter.”</p>
</div>
<div class="section level3">
<h3 id="convert-gene-annotations">Convert gene annotations<a class="anchor" aria-label="anchor" href="#convert-gene-annotations"></a>
</h3>
<p><img src="img/summary_06.png" width="400"></p>
<p>The SCTK can use annotation data from Bioconductor annotation packages such as <a href="https://www.bioconductor.org/packages/release/data/annotation/html/org.Hs.eg.db.html" class="external-link">org.Hs.eg.db</a> to convert gene annotations between standard annotation types such as Entrez gene ids, Ensembl gene ids, or gene symbols. By default, these packages are not installed, so you will need to manually install the package for your appropriate organism. Instructions for installation can be found on the <a href="https://www.bioconductor.org/install/#install-bioconductor-packages" class="external-link">Bioconductor website</a></p>
</div>
<div class="section level3">
<h3 id="delete-an-annotation-column">Delete an annotation column<a class="anchor" aria-label="anchor" href="#delete-an-annotation-column"></a>
</h3>
<p><img src="img/summary_07.png" width="400"></p>
<p>You can remove an unnecessary or unwanted annotation column by selecting it and clicking the “Delete Column” button.</p>
</div>
<div class="section level3">
<h3 id="randomly-subset">Randomly Subset<a class="anchor" aria-label="anchor" href="#randomly-subset"></a>
</h3>
<p><img src="img/summary_08.png" width="400"></p>
<p>If you are working with a large dataset, you may want to randomly subset the data for exploratory analysis. Specify the number of cells that you wish to keep in the subset data frame and click the “Subset Data” button.</p>
</div>
<div class="section level3">
<h3 id="download-scexperiment">Download SCExperiment<a class="anchor" aria-label="anchor" href="#download-scexperiment"></a>
</h3>
<p><img src="img/summary_09.png" width="400"></p>
<p>To export the SCExperiment object from the SCTK, you can download the object in RDS format by clicking the “Download SCExperiment” button. This file can be loaded into the R console for downstream analysis or for later upload and use in the SCTK.</p>
</div>
</div>
<div class="section level2">
<h2 id="assay-normalization">Assay Normalization<a class="anchor" aria-label="anchor" href="#assay-normalization"></a>
</h2>
<p>The assay normalization tab allows the user to conduct simple and complex normalization procedures on the assays in the SCTK experiment object. These include log and counts-per-million transforms, as well as library size normalization by scater and scran. It also lists the assays (raw and normalized) that are available in the object.</p>
<div class="section level3">
<h3 id="available-assays-and-basic-normalization">Available assays and basic normalization<a class="anchor" aria-label="anchor" href="#available-assays-and-basic-normalization"></a>
</h3>
<p><img src="img/summary_10.png" width="400"></p>
<p>The assays listed on this tab are stored in the underlying SCTK experiment object and are available on the tabs in the app. Users can add a log(counts), CPM (counts per million), and log(CPM) matrix directly through the app.</p>
</div>
<div class="section level3">
<h3 id="library-size-normalization">Library size normalization<a class="anchor" aria-label="anchor" href="#library-size-normalization"></a>
</h3>
<p><img src="img/summary_11.png" width="200"></p>
<p>Users can also conduct library size normalization using the scater and scran appraoches.</p>
</div>
</div>
<div class="section level2">
<h2 id="annotation-data">Annotation Data<a class="anchor" aria-label="anchor" href="#annotation-data"></a>
</h2>
<p>The annotation tab displays the annotation matrix.</p>
<p><img src="img/summary_12.png" width="600"></p>
<div class="section level3">
<h3 id="replace-annotation-data">Replace Annotation Data<a class="anchor" aria-label="anchor" href="#replace-annotation-data"></a>
</h3>
<p>To modify the annotation data frame, download the annotation data using the “Download Annotation Data” button, modify the data using a text editor or Excel, save the data as a .csv file, and re-upload the data in the “Upload and replace the annotation data” field.</p>
</div>
<div class="section level3">
<h3 id="batch-correction">Batch Correction<a class="anchor" aria-label="anchor" href="#batch-correction"></a>
</h3>
<p>Note that Batch correction has its own vignette and link: <a href="https://compbiomed.github.io/sctk_docs/articles/v06-tab04_Batch-Correction.html" class="external-link">Batch Correction Tab</a></p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] BiocStyle_2.22.0</span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] rstudioapi_0.13     knitr_1.38          magrittr_2.0.3     </span>
<span class="co">##  [4] R6_2.5.1            ragg_1.2.2          rlang_1.0.2        </span>
<span class="co">##  [7] fastmap_1.1.0       stringr_1.4.0       tools_4.1.2        </span>
<span class="co">## [10] xfun_0.30           cli_3.2.0           jquerylib_0.1.4    </span>
<span class="co">## [13] systemfonts_1.0.4   htmltools_0.5.2     yaml_2.3.5         </span>
<span class="co">## [16] digest_0.6.29       rprojroot_2.0.3     pkgdown_2.0.2      </span>
<span class="co">## [19] bookdown_0.25       textshaping_0.3.6   BiocManager_1.30.16</span>
<span class="co">## [22] purrr_0.3.4         sass_0.4.1          fs_1.5.2           </span>
<span class="co">## [25] memoise_2.0.1       cachem_1.0.6        evaluate_0.15      </span>
<span class="co">## [28] rmarkdown_2.14      stringi_1.7.6       compiler_4.1.2     </span>
<span class="co">## [31] bslib_0.3.1         desc_1.4.1          jsonlite_1.8.0</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
