<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Dimensionality Reduction and Filtering Tab • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Dimensionality Reduction and Filtering Tab">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_QC-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">3: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">4: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wevanjohnson/singleCellTK" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Dimensionality Reduction and Filtering Tab</h1>
            <h3 data-toc-skip class="subtitle">Single Cell Toolkit</h3>
                        <h4 data-toc-skip class="author">David Jenkins</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA;Program in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:dfj@bu.edu" class="email">dfj@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Mohammed Muzamil Khan</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA;Program in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:mmkhan@bu.edu" class="email">mmkhan@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Tyler Faits</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MAProgram in Bioinformatics, Boston University, Boston, MA<br><h4 data-toc-skip class="author">W. Evan Johnson</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MAProgram in Bioinformatics, Boston University, Boston, MA<br><small class="dont-index">Source: <a href="https://github.com/wevanjohnson/singleCellTK/blob/HEAD/vignettes/v05-tab03_Visualization-and-Clustering.Rmd" class="external-link"><code>vignettes/v05-tab03_Visualization-and-Clustering.Rmd</code></a></small>
      <div class="hidden name"><code>v05-tab03_Visualization-and-Clustering.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Visualization of scRNA-Seq data is crucial for identifying cell identifies as well as learning functional subclusters of cells present in the data. The visualization and clustering tab can be used to visualize scRNA-Seq data at the gene and sample level, and to reduce dimensions and cluster data for downstream analysis. The tab has 3 subtabs, Gene visualization, cell-wise visualization and clustering, and Celda. In some cases the values for these tabs can be precomputed, and the results will be stored inside the object. For this tab, especially for the cell-wise visualization, analysis results can be stored in the reducedDim slot of the SingleCellExperiment object. For example large datasets, PCA and t-SNE can take a long time, so pre-computing this data is recommended. If you wish to re-run PCA or t-SNE, you can click the “Re-run” button and the data will be replaced.</p>
</div>
<div class="section level2">
<h2 id="gene-visualization">Gene Visualization<a class="anchor" aria-label="anchor" href="#gene-visualization"></a>
</h2>
<p>The Gene-wise Visualization subtab helps users to visualize gene expression data versus annotation data for one or multiple genes of interest using either a boxplot, scatterplot, barplot, or heatmap. The types of plots available depends on the nature of the annotation variable of interest (e.g. factor data, numeric values, etc.). This can be helpful for visualizing individual genes of interest, including cell-identifying marker genes, or for displaying housekeeping genes for sample quality control or for quantifying housekeeping genes or artificial spike-in controls for sequencing quality control.</p>
<p><img src="img/geneVis.png"></p>
</div>
<div class="section level2">
<h2 id="cellwise-visualization-and-clustering">Cellwise Visualization and Clustering<a class="anchor" aria-label="anchor" href="#cellwise-visualization-and-clustering"></a>
</h2>
<p>The Cellwise Visualization and Clustering subtab includes dimensionality reduction and clustering techniques to allow users to visualize their scRNA-Seq in lower dimensional space. For example, the SCTK uses principal component analysis (PCA) to transform the expression matrix into orthogonal, linear dimensions that increasing describe the variation observed in the data. Furthermore, the SCTK can use popular non-linear dimension reduction methods such t-SNE and UMAP. We note that by default, the t-SNE and UMAP functions operate on a subset of the PCA-reduced data but can be applied on the raw data if desired.</p>
<div class="section level3">
<h3 id="dimension-reduction">Dimension reduction<a class="anchor" aria-label="anchor" href="#dimension-reduction"></a>
</h3>
<p>The Cell-wise Visualization and Clustering subtab in the SCTK provides a list of available (pre-computed) reduced dimension datasets available in the SingleCellExperiment object (if any are available). The user can select an assay matrix (e.g. counts, log-counts, counts per million, etc), and interactively apply one of the dimension reduction algorithms on the dataset (e.g. PCA, t-SNE, UMAP). For smaller datasets, users can perform PCA, t-SNE, and UMAP directly through the SCTK Shiny app. The resulting reduced data matrices will be stored in the underlying object for future use and downstream visualization, and will be available when the data object is downloaded (or re-uploaded). Because some of these algorithms can take a long time to compute on large datasets (e.g.  t-SNE on &gt;5,000 cells), users can precompute these reduced dimension data from the console, on the cluster, or in a preprocessing workflow, and store the reduced dimension results in the SingleCellExperiment object before uploading the object into the SCTK user interface.</p>
<p><img src="img/cellVisClust2.png"></p>
</div>
<div class="section level3">
<h3 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h3>
<p>The resulting reduced dimension data can be efficiently displayed in the visualization box. Annotation information can be added to the plot by selecting annotations with which to color or shape the points in the scatterplot.</p>
<p><img src="img/cellVisClust3.png"></p>
</div>
<div class="section level3">
<h3 id="clustering">Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h3>
<p>After visualization of the data, users may want to stratify the scRNA-Seq data into clusters that appear during dimensionality reduction. Users can choose to cluster their data using k-means clustering, hierarchical clustering, or CLARA (Clustering for Large Applications). After the clustering algorithm is complete, the visualization plot can be updated to to display the resulting clusters. The cluster results, the cluster assignments are stored in the annotation data frame (colData) of the SingleCellExperiment object. The clusters can then be used to filter the data for downstream visualization (e.g. differential expression within one or more clusters/cell types).</p>
<p><img src="img/cellVisClust4.png"></p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] BiocStyle_2.22.0</span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] rstudioapi_0.13     knitr_1.38          magrittr_2.0.3     </span>
<span class="co">##  [4] R6_2.5.1            ragg_1.2.2          rlang_1.0.2        </span>
<span class="co">##  [7] fastmap_1.1.0       stringr_1.4.0       tools_4.1.2        </span>
<span class="co">## [10] xfun_0.30           cli_3.2.0           jquerylib_0.1.4    </span>
<span class="co">## [13] systemfonts_1.0.4   htmltools_0.5.2     yaml_2.3.5         </span>
<span class="co">## [16] digest_0.6.29       rprojroot_2.0.3     pkgdown_2.0.2      </span>
<span class="co">## [19] bookdown_0.25       textshaping_0.3.6   BiocManager_1.30.16</span>
<span class="co">## [22] purrr_0.3.4         sass_0.4.1          fs_1.5.2           </span>
<span class="co">## [25] memoise_2.0.1       cachem_1.0.6        evaluate_0.15      </span>
<span class="co">## [28] rmarkdown_2.14      stringi_1.7.6       compiler_4.1.2     </span>
<span class="co">## [31] bslib_0.3.1         desc_1.4.1          jsonlite_1.8.0</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
