<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Differential Expression Tab • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Differential Expression Tab">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_QC-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">3: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">4: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wevanjohnson/singleCellTK" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Differential Expression Tab</h1>
            <h3 data-toc-skip class="subtitle">Single Cell Toolkit</h3>
                        <h4 data-toc-skip class="author">David Jenkins</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA;Program in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:dfj@bu.edu" class="email">dfj@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Mohammed Muzamil Khan</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA;Program in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:mmkhan@bu.edu" class="email">mmkhan@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Tyler Faits</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MAProgram in Bioinformatics, Boston University, Boston, MA<br><h4 data-toc-skip class="author">W. Evan Johnson</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MAProgram in Bioinformatics, Boston University, Boston, MA<br><small class="dont-index">Source: <a href="https://github.com/wevanjohnson/singleCellTK/blob/HEAD/vignettes/v07-tab05_Differential-Expression.Rmd" class="external-link"><code>vignettes/v07-tab05_Differential-Expression.Rmd</code></a></small>
      <div class="hidden name"><code>v07-tab05_Differential-Expression.Rmd</code></div>

    </address>
</address>
</div>

    
    
<p><img src="img/diffex_01.png"></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Differential expression analysis can identify genes that are significantly up or down regulated between conditions. While many differential expression algorithms exist, their performance may vary on scRNA-Seq datasets.</p>
</div>
<div class="section level2">
<h2 id="limma-deseq2-and-anova">limma, DESeq2, and ANOVA<a class="anchor" aria-label="anchor" href="#limma-deseq2-and-anova"></a>
</h2>
<p>Users can apply common differential expression algorithms <a href="https://bioconductor.org/packages/release/bioc/html/limma.html" class="external-link">limma</a>, <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" class="external-link">DESeq2</a>, or perform an ANOVA to identify differentially expressed genes by selecting one or multiple condition variables present in the annotation information. After choosing the assay of interest using the “Select Assay” field and the algorithm in the “Select Method” field, choose the experimental condition and any additional covariates to use in the differential expression model. The resulting genes will be differentially expressed between the factor of interest and all other cells. Users can also add addition covariates to control for in the model and customize the differential expression results by changing p-value correction method applied to the results.</p>
<div class="section level3">
<h3 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h3>
<p><img src="img/diffex_02.png"></p>
<p>The resulting gene list is displayed as a table in the “Results Table” tab and also in a heatmap in the “Heatmap” tab, which can also be customized using the options available in the “Options” tab. Users can download the gene list directly or create a biomarker list for a specific cell type or cell cluster, which can be stored in the gene annotation information in the SCExperiment object.</p>
</div>
</div>
<div class="section level2">
<h2 id="mast">MAST<a class="anchor" aria-label="anchor" href="#mast"></a>
</h2>
<p><img src="img/diffex_04.png"></p>
<p><a href="https://www.bioconductor.org/packages/release/bioc/html/MAST.html" class="external-link">MAST</a>, Model-based Analysis of Single-cell Transcriptomics, is a differential expression analysis tool specifically designed for single cell RNA-Seq data, which uses a hurdle model to account for the missingness in scRNA-Seq data. MAST has been implemented within the SCTK. Users can choose whether to use MAST’s adaptive thresholding model, choose fold change and expression thresholds, and identify significant genes based on conditions present in the annotation information provided. The results are presented in a table, violin plots, or visualized in a heatmap. For detailed information about MAST analysis, see the <a href="https://www.bioconductor.org/packages/release/bioc/html/MAST.html" class="external-link">MAST documentation</a>.</p>
<p><img src="img/diffex_05.png"></p>
<p><img src="img/diffex_06.png"></p>
</div>
<div class="section level2">
<h2 id="saving-and-downloading-results">Saving and Downloading Results<a class="anchor" aria-label="anchor" href="#saving-and-downloading-results"></a>
</h2>
<p>After differential expression analysis has been performed, there are several convenient options for downstream usage of the results. If desired, the entire resulting analysis can be stored in the SCExperiment object for future use. The user can access these results in the “Saved Results” button at the top of the “Options” panel.</p>
<p><img src="img/diffex_07.png" style="width:50.0%"></p>
<p>The user can also store the differential gene list, “Save to ‘N’ Signficant genes” for additional analysis or pathway enrichment analysis in other parts of the SCTK.</p>
<p><img src="img/diffex_08.png" style="width:50.0%"></p>
<p>Finally, the results can be downloaded in .csv form for viewing and use outside the SCTK and outside of R.<br><img src="img/diffex_09.png" style="width:50.0%"></p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] BiocStyle_2.22.0</span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] rstudioapi_0.13     knitr_1.38          magrittr_2.0.3     </span>
<span class="co">##  [4] R6_2.5.1            ragg_1.2.2          rlang_1.0.2        </span>
<span class="co">##  [7] fastmap_1.1.0       stringr_1.4.0       tools_4.1.2        </span>
<span class="co">## [10] xfun_0.30           cli_3.2.0           jquerylib_0.1.4    </span>
<span class="co">## [13] systemfonts_1.0.4   htmltools_0.5.2     yaml_2.3.5         </span>
<span class="co">## [16] digest_0.6.29       rprojroot_2.0.3     pkgdown_2.0.2      </span>
<span class="co">## [19] bookdown_0.25       textshaping_0.3.6   BiocManager_1.30.16</span>
<span class="co">## [22] purrr_0.3.4         sass_0.4.1          fs_1.5.2           </span>
<span class="co">## [25] memoise_2.0.1       cachem_1.0.6        evaluate_0.15      </span>
<span class="co">## [28] rmarkdown_2.14      stringi_1.7.6       compiler_4.1.2     </span>
<span class="co">## [31] bslib_0.3.1         desc_1.4.1          jsonlite_1.8.0</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
