<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Sample Size Tab • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Sample Size Tab">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_QC-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">3: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">4: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wevanjohnson/singleCellTK" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Sample Size Tab</h1>
            <h3 data-toc-skip class="subtitle">Single Cell Toolkit</h3>
                        <h4 data-toc-skip class="author">David Jenkins</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MAProgram in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:dfj@bu.edu" class="email">dfj@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Tyler Faits</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MAProgram in Bioinformatics, Boston University, Boston, MA<br><h4 data-toc-skip class="author">W. Evan Johnson</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MAProgram in Bioinformatics, Boston University, Boston, MA<br><small class="dont-index">Source: <a href="https://github.com/wevanjohnson/singleCellTK/blob/HEAD/vignettes/v09-tab07_Sample-Size.Rmd" class="external-link"><code>vignettes/v09-tab07_Sample-Size.Rmd</code></a></small>
      <div class="hidden name"><code>v09-tab07_Sample-Size.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The Sample Size tab allows users to create simulated datasets built from their real data, and to perform power calculations to help guide future experimental designs. The subsampler will display summary statistics for number of detected genes, number of significantly differentially expressed genes, and minimum detectable effect sizes for simulated datasets of varying design.</p>
<div class="section level3">
<h3 id="downsample-depth">Downsample Depth<a class="anchor" aria-label="anchor" href="#downsample-depth"></a>
</h3>
<p>In the ‘Sequencing Depth’ tab, you can see how the statistical power of simulated experiments changes as the number of sequencing reads is changed. You can specify the minimum and maximum number of reads to simulate, select whether those parameters are total reads per experiment or reads per cell (default is total reads/experiment), and choose a condition for differential expression assesment from your sample annotations. The chosen condition for differential expression should be a discrete variable. For expediency, differential expression is performed using either a t-test if exactly two groups are detected for the chosen condition, or an ANOVA if more than two groups are detected. The toolkit will only run the simulations once you click the ‘Run Subsampler’ button.</p>
<p>Simulated datasets will not be permanently saved. Instead, summary statistics will be saved for each simulation, and plotted in the “Genes Detected”, “Minimum Detectable Effect Size”, and “Number of Diffex Genes” tabs. Solid lines represent median values at each simulated sequencing depth, wile dotted lines represent the upper and lower quartiles.</p>
<p><img src="img/samplesize_01.png"></p>
</div>
<div class="section level3">
<h3 id="downsample-cells">Downsample Cells<a class="anchor" aria-label="anchor" href="#downsample-cells"></a>
</h3>
<p>In the ‘Number of Cells’ tab, you can see how the statistical power of simulated experiments changes as the number of cells is changed while keeping the total number of reads for the experiment constant. You can specify the minimum and maximum number of cells to simulate, specify how many total reads each simulated counts matrix should have, and choose a condition for differential expression assesment from your sample annotations. The chosen condition for differential expression should be a discrete variable. For expediency, differential expression is performed using either a t-test if exactly two groups are detected for the chosen condition, or an ANOVA if more than two groups are detected. The toolkit will only run the simulations once you click the ‘Run Subsampler’ button.</p>
<p>Simulated datasets will not be permanently saved. Instead, summary statistics will be saved for each simulation, and plotted in the “Genes Detected”, “Minimum Detectable Effect Size”, and “Number of Diffex Genes” tabs. Solid lines represent median values at each simulated sequencing depth, wile dotted lines represent the upper and lower quartiles.</p>
<p><img src="img/samplesize_02.png"></p>
</div>
<div class="section level3">
<h3 id="snapshot">Snapshot<a class="anchor" aria-label="anchor" href="#snapshot"></a>
</h3>
<p>In the ‘Snapshot’ tab, you can simultaneously specify both the number of cells and the total number of reads to be produced in each simulated experiment. Rather than giving you an impression of how statistical power would be affected by a variable experimental design, this tab shows a more detailed assesment of a single proposed design.</p>
<p>In the snapshot plot, we show a scatterplot of all genes detected in your input dataset. The x-axis represents the empirical cohen’s-d effect size for genes; cohen’s-d is defined as the mean expression level for a gene in condition A minus the mean expression level in condition B, divided by the pooled standard deviation in expression of that gene across all cells. If your chosen condition of interest has more than two groups, the largest two groups will be compared. The y-axis shows the portion of simulated datasets in which a gene was found to be significantly differentially expressed between groups. This should be interpreted as an estimate of statistical power.</p>
<p><img src="img/samplesize_03.png"></p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] BiocStyle_2.22.0</span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] rstudioapi_0.13     knitr_1.38          magrittr_2.0.3     </span>
<span class="co">##  [4] R6_2.5.1            ragg_1.2.2          rlang_1.0.2        </span>
<span class="co">##  [7] fastmap_1.1.0       stringr_1.4.0       tools_4.1.2        </span>
<span class="co">## [10] xfun_0.30           cli_3.2.0           jquerylib_0.1.4    </span>
<span class="co">## [13] systemfonts_1.0.4   htmltools_0.5.2     yaml_2.3.5         </span>
<span class="co">## [16] digest_0.6.29       rprojroot_2.0.3     pkgdown_2.0.2      </span>
<span class="co">## [19] bookdown_0.25       textshaping_0.3.6   BiocManager_1.30.16</span>
<span class="co">## [22] purrr_0.3.4         sass_0.4.1          fs_1.5.2           </span>
<span class="co">## [25] memoise_2.0.1       cachem_1.0.6        evaluate_0.15      </span>
<span class="co">## [28] rmarkdown_2.14      stringi_1.7.6       compiler_4.1.2     </span>
<span class="co">## [31] bslib_0.3.1         desc_1.4.1          jsonlite_1.8.0</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
