<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Align Single Cell RNA-Seq Data and Create a SCtkExperiment Object — alignSingleCellData • singleCellTK</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Align Single Cell RNA-Seq Data and Create a SCtkExperiment Object — alignSingleCellData"><meta property="og:description" content="Align Single Cell RNA-Seq Data and Create a SCtkExperiment Object"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_QC-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">3: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">4: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/wevanjohnson/singleCellTK" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Align Single Cell RNA-Seq Data and Create a SCtkExperiment Object</h1>
    <small class="dont-index">Source: <a href="https://github.com/wevanjohnson/singleCellTK/blob/HEAD/R/alignSingleCellData.R" class="external-link"><code>R/alignSingleCellData.R</code></a></small>
    <div class="hidden name"><code>alignSingleCellData.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Align Single Cell RNA-Seq Data and Create a SCtkExperiment Object</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">alignSingleCellData</span><span class="op">(</span>
  <span class="va">inputfile1</span>,
  inputfile2 <span class="op">=</span> <span class="cn">NULL</span>,
  <span class="va">indexPath</span>,
  <span class="va">gtfAnnotation</span>,
  outputDir <span class="op">=</span> <span class="cn">NULL</span>,
  sampleAnnotations <span class="op">=</span> <span class="cn">NULL</span>,
  featureAnnotations <span class="op">=</span> <span class="cn">NULL</span>,
  threads <span class="op">=</span> <span class="fl">1</span>,
  saveBam <span class="op">=</span> <span class="cn">FALSE</span>,
  saveCountFiles <span class="op">=</span> <span class="cn">FALSE</span>,
  isPairedEnd <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>inputfile1</dt>
<dd><p>An input file or list of files. Files can be fastq,
fastq.gz, or bam, but must all be of the same type. Sample names will be the
full file name, without _1.fastq.gz, .fastq.gz, _1.fastq, .fastq or .bam
endings.</p></dd>
<dt>inputfile2</dt>
<dd><p>If fastq files are provided in input list, a list of
corresponding paired fastq files, if applicable.</p></dd>
<dt>indexPath</dt>
<dd><p>Path to the Rsubread genome index.</p></dd>
<dt>gtfAnnotation</dt>
<dd><p>Path to the GTF gene annotation to use. This must
correspond to the genome specified in indexPath.</p></dd>
<dt>outputDir</dt>
<dd><p>If saveBam or saveCountFiles is TRUE, specify a
directory in which to save the output files.</p></dd>
<dt>sampleAnnotations</dt>
<dd><p>A data.frame of sample annotations, with samples
as rows and annotations in columns. The sample names must be identical to
and in the same order as the list of files in inputfile1. Alignment
statistics will be added to the annotation data frame.</p></dd>
<dt>featureAnnotations</dt>
<dd><p>An optional data.frame of probe annotations, with
probes as rows and probe annotations in columns.</p></dd>
<dt>threads</dt>
<dd><p>Number of threads to use during alignment. The default is 1.</p></dd>
<dt>saveBam</dt>
<dd><p>If TRUE, bam alignment files will be saved in the outputDir.
The default is FALSE.</p></dd>
<dt>saveCountFiles</dt>
<dd><p>If TRUE, per sample gene count files will be saved in
the outputDir. The default is FALSE.</p></dd>
<dt>isPairedEnd</dt>
<dd><p>If input files are .bam, indicate whether the input bam
files are paired end.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Object to import into the shiny app.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in"><span class="va">singlecellobject</span> <span class="op">&lt;-</span> <span class="fu">alignSingleCellData</span><span class="op">(</span></span>
<span class="r-in">  inputfile1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"/path/to/sample1_1.fastq.gz"</span>,</span>
<span class="r-in">                 <span class="st">"/path/to/sample2_1.fastq.gz"</span><span class="op">)</span>,</span>
<span class="r-in">  inputfile2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"/path/to/sample1_2.fastq.gz"</span>,</span>
<span class="r-in">                 <span class="st">"/path/to/sample2_2.fastq.gz"</span><span class="op">)</span>,</span>
<span class="r-in">  indexPath <span class="op">=</span> <span class="st">"/path/to/genome/index"</span>,</span>
<span class="r-in">  gtfAnnotation <span class="op">=</span> <span class="st">"/path/to/gene/annotations.gtf"</span>,</span>
<span class="r-in">  sampleAnnotations <span class="op">=</span> <span class="va">sample.annotation.df</span>,</span>
<span class="r-in">  threads<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">}</span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

