<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Run GSVA analysis on a SCtkExperiment object. — gsvaSCE • singleCellTK</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Run GSVA analysis on a SCtkExperiment object. — gsvaSCE"><meta property="og:description" content="Run GSVA analysis on a SCtkExperiment object."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_QC-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">3: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">4: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/wevanjohnson/singleCellTK" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run GSVA analysis on a SCtkExperiment object.</h1>
    <small class="dont-index">Source: <a href="https://github.com/wevanjohnson/singleCellTK/blob/HEAD/R/GSVA.R" class="external-link"><code>R/GSVA.R</code></a></small>
    <div class="hidden name"><code>gsvaSCE.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Run GSVA analysis on a SCtkExperiment object.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">gsvaSCE</span><span class="op">(</span><span class="va">inSCE</span>, useAssay <span class="op">=</span> <span class="st">"logcounts"</span>, <span class="va">pathwaySource</span>, <span class="va">pathwayNames</span>, <span class="va">...</span><span class="op">)</span>

<span class="fu">gsvaPlot</span><span class="op">(</span>
  <span class="va">inSCE</span>,
  <span class="va">gsvaData</span>,
  <span class="va">plotType</span>,
  condition <span class="op">=</span> <span class="cn">NULL</span>,
  show_column_names <span class="op">=</span> <span class="cn">TRUE</span>,
  show_row_names <span class="op">=</span> <span class="cn">TRUE</span>,
  text_size <span class="op">=</span> <span class="fl">12</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>inSCE</dt>
<dd><p>Input SCtkExperiment object. Required</p></dd>
<dt>useAssay</dt>
<dd><p>Indicate which assay to use. The default is "logcounts"</p></dd>
<dt>pathwaySource</dt>
<dd><p>The pathway source if "Manual Input", the pathwayNames
should be rowData annotations that are (0,1) vectors. If, "MSigDB c2 (Human,
Entrez ID only)", the pathwayNames should be pathways from MSigDB c2 or "ALL"
to run on all available pathways.</p></dd>
<dt>pathwayNames</dt>
<dd><p>List of pathway names to run, depending on pathwaySource
parameter.</p></dd>
<dt>...</dt>
<dd><p>Parameters to pass to gsva()</p></dd>
<dt>gsvaData</dt>
<dd><p>GSVA data to plot. Required.</p></dd>
<dt>plotType</dt>
<dd><p>The type of plot to use, "Violin" or "Heatmap". Required.</p></dd>
<dt>condition</dt>
<dd><p>The condition(s) to use for the Violin plot, or the
condition(s) to add as color bars above the Heatmap. Required for Violin,
optional for Heatmap.</p></dd>
<dt>show_column_names</dt>
<dd><p>Display the column labels on the heatmap. The
default is TRUE</p></dd>
<dt>show_row_names</dt>
<dd><p>Display the row labels on the heatmap. The default
is TRUE.</p></dd>
<dt>text_size</dt>
<dd><p>Text size for plots. The default is 12</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>gsvaSCE(): A data.frame of pathway activity scores from GSVA.
gsvaPlot(): The requested plot of the GSVA results.</p>
    </div>
    <div id="functions">
    <h2>Functions</h2>
    
<ul><li><p><code>gsvaPlot</code>: Plot GSVA results.</p>
<p>Plot GSVA Results</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">maits</span>, package <span class="op">=</span> <span class="st">"MAST"</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">c2BroadSets</span>, package <span class="op">=</span> <span class="st">"GSVAdata"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">maitslogtpm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DelayedArray/man/DelayedMatrix-utils.html" class="external-link">t</a></span><span class="op">(</span><span class="va">maits</span><span class="op">$</span><span class="va">expressionmat</span><span class="op">)</span></span>
<span class="r-in"><span class="va">genesToSubset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">maitslogtpm</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">maitslogtpm</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span></span>
<span class="r-in">                 <span class="fu">GSEABase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">c2BroadSets</span><span class="op">[[</span><span class="st">"KEGG_PROTEASOME"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span class="r-in"><span class="va">maitslogtpm</span> <span class="op">&lt;-</span> <span class="va">maitslogtpm</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">maitslogtpm</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">genesToSubset</span>, <span class="op">]</span></span>
<span class="r-in"><span class="va">maitsfeatures</span> <span class="op">&lt;-</span> <span class="va">maits</span><span class="op">$</span><span class="va">fdat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">maits</span><span class="op">$</span><span class="va">fdat</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">genesToSubset</span>, <span class="op">]</span></span>
<span class="r-in"><span class="va">maitsSCE</span> <span class="op">&lt;-</span> <span class="fu"><a href="createSCE.html">createSCE</a></span><span class="op">(</span>assayFile <span class="op">=</span> <span class="va">maitslogtpm</span>, annotFile <span class="op">=</span> <span class="va">maits</span><span class="op">$</span><span class="va">cdat</span>,</span>
<span class="r-in">                      featureFile <span class="op">=</span> <span class="va">maitsfeatures</span>, assayName <span class="op">=</span> <span class="st">"logtpm"</span>,</span>
<span class="r-in">                      inputDataFrames <span class="op">=</span> <span class="cn">TRUE</span>, createLogCounts <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">maitsSCE</span><span class="op">)</span><span class="op">$</span><span class="va">testbiomarker</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">maitsSCE</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">gsvaSCE</span><span class="op">(</span>inSCE <span class="op">=</span> <span class="va">maitsSCE</span>, useAssay <span class="op">=</span> <span class="st">"logtpm"</span>,</span>
<span class="r-in">               pathwaySource <span class="op">=</span> <span class="st">"Manual Input"</span>, pathwayNames <span class="op">=</span> <span class="st">"testbiomarker"</span>,</span>
<span class="r-in">               parallel.sz <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating GSVA scores for 1 gene sets.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating ECDFs with Gaussian kernels</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="co">#Create a small example to run</span></span>
<span class="r-in"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">maits</span>, package <span class="op">=</span> <span class="st">"MAST"</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">c2BroadSets</span>, package <span class="op">=</span> <span class="st">"GSVAdata"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">maitslogtpm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DelayedArray/man/DelayedMatrix-utils.html" class="external-link">t</a></span><span class="op">(</span><span class="va">maits</span><span class="op">$</span><span class="va">expressionmat</span><span class="op">)</span></span>
<span class="r-in"><span class="va">genesToSubset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">maitslogtpm</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">maitslogtpm</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span></span>
<span class="r-in">                 <span class="fu">GSEABase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">c2BroadSets</span><span class="op">[[</span><span class="st">"KEGG_PROTEASOME"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span class="r-in"><span class="va">maitslogtpm</span> <span class="op">&lt;-</span> <span class="va">maitslogtpm</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">maitslogtpm</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">genesToSubset</span>, <span class="op">]</span></span>
<span class="r-in"><span class="va">maitsfeatures</span> <span class="op">&lt;-</span> <span class="va">maits</span><span class="op">$</span><span class="va">fdat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">maits</span><span class="op">$</span><span class="va">fdat</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">genesToSubset</span>, <span class="op">]</span></span>
<span class="r-in"><span class="va">maitsSCE</span> <span class="op">&lt;-</span> <span class="fu"><a href="createSCE.html">createSCE</a></span><span class="op">(</span>assayFile <span class="op">=</span> <span class="va">maitslogtpm</span>, annotFile <span class="op">=</span> <span class="va">maits</span><span class="op">$</span><span class="va">cdat</span>,</span>
<span class="r-in">                      featureFile <span class="op">=</span> <span class="va">maitsfeatures</span>, assayName <span class="op">=</span> <span class="st">"logtpm"</span>,</span>
<span class="r-in">                      inputDataFrames <span class="op">=</span> <span class="cn">TRUE</span>, createLogCounts <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">maitsSCE</span><span class="op">)</span><span class="op">$</span><span class="va">testbiomarker</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">maitsSCE</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">gsvaSCE</span><span class="op">(</span>inSCE <span class="op">=</span> <span class="va">maitsSCE</span>, useAssay <span class="op">=</span> <span class="st">"logtpm"</span>,</span>
<span class="r-in">               pathwaySource <span class="op">=</span> <span class="st">"Manual Input"</span>, pathwayNames <span class="op">=</span> <span class="st">"testbiomarker"</span>,</span>
<span class="r-in">               parallel.sz <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating GSVA scores for 1 gene sets.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating ECDFs with Gaussian kernels</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="fu">gsvaPlot</span><span class="op">(</span>inSCE <span class="op">=</span> <span class="va">maitsSCE</span>, gsvaData <span class="op">=</span> <span class="va">res</span>,</span>
<span class="r-in">         plotType <span class="op">=</span> <span class="st">"Violin"</span>, condition <span class="op">=</span> <span class="st">"condition"</span><span class="op">)</span></span>
<span class="r-plt img"><img src="gsvaSCE-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

