<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ singleCellTK</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_QC-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">3: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">4: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/wevanjohnson/singleCellTK" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="shiny-app">Shiny App <a href="#shiny-app" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="singleCellTK.html">singleCellTK()</a></code> </p>
        </td>
        <td><p>Run the single cell analysis app</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="data-summary-and-filtering">Data Summary and Filtering <a href="#data-summary-and-filtering" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="filterSCData.html">filterSCData()</a></code> </p>
        </td>
        <td><p>Filter Genes and Samples from a Single Cell Object</p></td>
      </tr><tr><td>
          <p><code><a href="summarizeTable.html">summarizeTable()</a></code> </p>
        </td>
        <td><p>Summarize SCtkExperiment</p></td>
      </tr><tr><td>
          <p><code><a href="convertGeneIDs.html">convertGeneIDs()</a></code> </p>
        </td>
        <td><p>Convert Gene IDs</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="dimensionality-reduction-and-clustering">Dimensionality Reduction and Clustering <a href="#dimensionality-reduction-and-clustering" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="getPCA.html">getPCA()</a></code> </p>
        </td>
        <td><p>Get and plot PCA components for a SCtkE object</p></td>
      </tr><tr><td>
          <p><code><a href="getTSNE.html">getTSNE()</a></code> </p>
        </td>
        <td><p>Run t-SNE dimensionality reduction method on the assay data.</p></td>
      </tr><tr><td>
          <p><code><a href="getBiomarker.html">getBiomarker()</a></code> </p>
        </td>
        <td><p>Given a list of genes and a SCtkExperiment object, return
the binary or continuous expression of the genes.</p></td>
      </tr><tr><td>
          <p><code><a href="getClusterInputData.html">getClusterInputData()</a></code> </p>
        </td>
        <td><p>Get data to use as input clustering algorithms</p></td>
      </tr><tr><td>
          <p><code><a href="plotBiomarker.html">plotBiomarker()</a></code> </p>
        </td>
        <td><p>Given a set of genes, return a ggplot of expression
values.</p></td>
      </tr><tr><td>
          <p><code><a href="plotPCA.html">plotPCA()</a></code> </p>
        </td>
        <td><p>Plot PCA run data from its components.</p></td>
      </tr><tr><td>
          <p><code><a href="plotTSNE.html">plotTSNE()</a></code> </p>
        </td>
        <td><p>Plot t-SNE plot on dimensionality reduction data run from t-SNE method.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="batch-correction">Batch Correction <a href="#batch-correction" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="plotBatchVariance.html">plotBatchVariance()</a></code> </p>
        </td>
        <td><p>Plot the percent of the variation that is explained by batch and condition
in the data</p></td>
      </tr><tr><td>
          <p><code><a href="ComBatSCE.html">ComBatSCE()</a></code> </p>
        </td>
        <td><p>ComBatSCE</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="differential-expression">Differential Expression <a href="#differential-expression" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="scDiffEx.html">scDiffEx()</a></code> <code><a href="scDiffEx.html">scDiffExDESeq2()</a></code> <code><a href="scDiffEx.html">scDiffExlimma()</a></code> <code><a href="scDiffEx.html">scDiffExANOVA()</a></code> </p>
        </td>
        <td><p>Perform differential expression analysis on a SCtkExperiment object</p></td>
      </tr><tr><td>
          <p><code><a href="plotDiffEx.html">plotDiffEx()</a></code> </p>
        </td>
        <td><p>Plot Differential Expression</p></td>
      </tr><tr><td>
          <p><code><a href="MAST.html">MAST()</a></code> <code><a href="MAST.html">thresholdGenes()</a></code> <code><a href="MAST.html">MASTviolin()</a></code> <code><a href="MAST.html">MASTregression()</a></code> </p>
        </td>
        <td><p>MAST</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="pathway-activity-analysis">Pathway Activity Analysis <a href="#pathway-activity-analysis" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="gsvaSCE.html">gsvaSCE()</a></code> <code><a href="gsvaSCE.html">gsvaPlot()</a></code> </p>
        </td>
        <td><p>Run GSVA analysis on a SCtkExperiment object.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="sample-size">Sample Size <a href="#sample-size" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="calcEffectSizes.html">calcEffectSizes()</a></code> </p>
        </td>
        <td><p>Finds the effect sizes for all genes in the original dataset, regardless of
significance.</p></td>
      </tr><tr><td>
          <p><code><a href="DownsampleCells.html">DownsampleCells()</a></code> </p>
        </td>
        <td><p>Estimate numbers of detected genes, significantly differentially expressed
genes, and median significant effect size</p></td>
      </tr><tr><td>
          <p><code><a href="DownsampleDepth.html">DownsampleDepth()</a></code> </p>
        </td>
        <td><p>Estimate numbers of detected genes, significantly differentially expressed
genes, and median significant effect size</p></td>
      </tr><tr><td>
          <p><code><a href="generateSimulatedData.html">generateSimulatedData()</a></code> </p>
        </td>
        <td><p>Generates a single simulated dataset, bootstrapping from the input counts
matrix.</p></td>
      </tr><tr><td>
          <p><code><a href="iterateSimulations.html">iterateSimulations()</a></code> </p>
        </td>
        <td><p>Returns significance data from a snapshot.</p></td>
      </tr><tr><td>
          <p><code><a href="subDiffEx.html">subDiffEx()</a></code> <code><a href="subDiffEx.html">subDiffExttest()</a></code> <code><a href="subDiffEx.html">subDiffExANOVA()</a></code> </p>
        </td>
        <td><p>Passes the output of generateSimulatedData() to differential expression
tests, picking either t-tests or ANOVA for data with only two conditions or
multiple conditions, respectively.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="data">Data <a href="#data" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="mouseBrainSubsetSCE.html">mouseBrainSubsetSCE</a></code> </p>
        </td>
        <td><p>Example Single Cell RNA-Seq data in SCtkExperiment Object, GSE60361
subset</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="sctkexperiment">SCtkExperiment <a href="#sctkexperiment" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="createSCE.html">createSCE()</a></code> </p>
        </td>
        <td><p>Create a SCtkExperiment object</p></td>
      </tr><tr><td>
          <p><code><a href="SCtkExperiment-class.html">SCtkExperiment-class</a></code> </p>
        </td>
        <td><p>A lightweight S4 extension to the SingleCellExperiment class to store
additional information.</p></td>
      </tr><tr><td>
          <p><code><a href="SCtkExperiment.html">SCtkExperiment()</a></code> </p>
        </td>
        <td><p>Create a SCtkExperiment</p></td>
      </tr><tr><td>
          <p><code><a href="pcaVariances-set.html">`pcaVariances&lt;-`()</a></code> </p>
        </td>
        <td><p>Set PCA variances</p></td>
      </tr><tr><td>
          <p><code><a href="pcaVariances.html">pcaVariances()</a></code> <code><a href="pcaVariances.html">`pcaVariances&lt;-`(<i>&lt;SCtkExperiment&gt;</i>)</a></code> </p>
        </td>
        <td><p>Get PCA variances</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="alignment">Alignment <a href="#alignment" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="alignSingleCellData.html">alignSingleCellData()</a></code> </p>
        </td>
        <td><p>Align Single Cell RNA-Seq Data and Create a SCtkExperiment Object</p></td>
      </tr><tr><td>
          <p><code><a href="parseRsubreadLogs.html">parseRsubreadLogs()</a></code> </p>
        </td>
        <td><p>Parse Rsubread Logs for Mapping and Feature Count Statistics</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

