<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Perform differential expression analysis on a SCtkExperiment object — scDiffEx • singleCellTK</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Perform differential expression analysis on a SCtkExperiment object — scDiffEx"><meta property="og:description" content="Perform differential expression analysis on a SCtkExperiment object"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_QC-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">3: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">4: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/wevanjohnson/singleCellTK" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform differential expression analysis on a SCtkExperiment object</h1>
    <small class="dont-index">Source: <a href="https://github.com/wevanjohnson/singleCellTK/blob/HEAD/R/scDiffEx.R" class="external-link"><code>R/scDiffEx.R</code></a></small>
    <div class="hidden name"><code>scDiffEx.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Perform differential expression analysis on a SCtkExperiment object</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">scDiffEx</span><span class="op">(</span>
  <span class="va">inSCE</span>,
  useAssay <span class="op">=</span> <span class="st">"logcounts"</span>,
  <span class="va">condition</span>,
  covariates <span class="op">=</span> <span class="cn">NULL</span>,
  significance <span class="op">=</span> <span class="fl">0.05</span>,
  ntop <span class="op">=</span> <span class="fl">500</span>,
  usesig <span class="op">=</span> <span class="cn">TRUE</span>,
  <span class="va">diffexmethod</span>,
  levelofinterest <span class="op">=</span> <span class="cn">NULL</span>,
  analysisType <span class="op">=</span> <span class="cn">NULL</span>,
  controlLevel <span class="op">=</span> <span class="cn">NULL</span>,
  adjust <span class="op">=</span> <span class="st">"fdr"</span>
<span class="op">)</span>

<span class="fu">scDiffExDESeq2</span><span class="op">(</span>
  <span class="va">inSCE</span>,
  useAssay <span class="op">=</span> <span class="st">"counts"</span>,
  <span class="va">condition</span>,
  analysisType <span class="op">=</span> <span class="st">"biomarker"</span>,
  levelofinterest <span class="op">=</span> <span class="cn">NULL</span>,
  controlLevel <span class="op">=</span> <span class="cn">NULL</span>,
  covariates <span class="op">=</span> <span class="cn">NULL</span>,
  adjust <span class="op">=</span> <span class="st">"fdr"</span>
<span class="op">)</span>

<span class="fu">scDiffExlimma</span><span class="op">(</span>
  <span class="va">inSCE</span>,
  useAssay <span class="op">=</span> <span class="st">"logcounts"</span>,
  <span class="va">condition</span>,
  analysisType <span class="op">=</span> <span class="st">"biomarker"</span>,
  levelofinterest <span class="op">=</span> <span class="cn">NULL</span>,
  covariates <span class="op">=</span> <span class="cn">NULL</span>,
  adjust <span class="op">=</span> <span class="st">"fdr"</span>
<span class="op">)</span>

<span class="fu">scDiffExANOVA</span><span class="op">(</span>
  <span class="va">inSCE</span>,
  useAssay <span class="op">=</span> <span class="st">"logcounts"</span>,
  <span class="va">condition</span>,
  covariates <span class="op">=</span> <span class="cn">NULL</span>,
  adjust <span class="op">=</span> <span class="st">"fdr"</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>inSCE</dt>
<dd><p>Input SCtkExperiment object. Required</p></dd>
<dt>useAssay</dt>
<dd><p>Indicate which assay to use. Default is "logcounts" for limma
and ANOVA, and "counts" for DESeq2.</p></dd>
<dt>condition</dt>
<dd><p>The name of the condition to use for differential
expression. Must be a name of a column from colData that contains at least
two labels. Required</p></dd>
<dt>covariates</dt>
<dd><p>Additional covariates to add to the model. Default is NULL</p></dd>
<dt>significance</dt>
<dd><p>FDR corrected significance cutoff for differentially
expressed genes. Required</p></dd>
<dt>ntop</dt>
<dd><p>Number of top differentially expressed genes to display in the
heatmap. Required</p></dd>
<dt>usesig</dt>
<dd><p>If TRUE, only display genes that meet the significance cutoff,
up to ntop genes. Required</p></dd>
<dt>diffexmethod</dt>
<dd><p>The method for performing differential expression
analysis. Available options are DESeq2, limma, and ANOVA. Required</p></dd>
<dt>levelofinterest</dt>
<dd><p>If the condition has more than two labels,
levelofinterest should contain one factor for condition. The differential
expression results will use levelofinterest depending on the analysisType
parameter.</p></dd>
<dt>analysisType</dt>
<dd><p>For conditions with more than two levels, limma and
DESeq2 can be run using multiple methods. For DESeq2, choose "biomarker" to
compare the levelofinterest to all other samples. Choose "contrast" to
compare the levelofinterest to a controlLevel (see below). Choose
"fullreduced" to perform DESeq2 in LRT mode. For limma, Choose "biomarker" to
compare the levelofinterest to all other samples. Choose "coef" to select a
coefficient of interest with levelofinterest (see below). Choose "allcoef" to
test if any coefficient is different from zero.</p></dd>
<dt>controlLevel</dt>
<dd><p>If the condition has more than two labels, controlLevel
should contain one factor from condition to use as the control.</p></dd>
<dt>adjust</dt>
<dd><p>Method for p-value correction. See options in p.adjust().
The default is fdr.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A data frame of gene names and adjusted p-values</p>
    </div>
    <div id="functions">
    <h2>Functions</h2>
    
<ul><li><p><code>scDiffExDESeq2</code>: Perform differential expression analysis with DESeq2</p></li>
<li><p><code>scDiffExlimma</code>: Perform differential expression analysis with limma</p></li>
<li><p><code>scDiffExANOVA</code>: Perform differential expression analysis with ANOVA</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"mouseBrainSubsetSCE"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">scDiffEx</span><span class="op">(</span><span class="va">mouseBrainSubsetSCE</span>,</span>
<span class="r-in">                useAssay <span class="op">=</span> <span class="st">"logcounts"</span>,</span>
<span class="r-in">                <span class="st">"level1class"</span>,</span>
<span class="r-in">                diffexmethod <span class="op">=</span> <span class="st">"limma"</span><span class="op">)</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Zero sample variances detected, have been offset away from zero</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"mouseBrainSubsetSCE"</span><span class="op">)</span></span>
<span class="r-in"><span class="co">#sort first 100 expressed genes</span></span>
<span class="r-in"><span class="va">ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">mouseBrainSubsetSCE</span><span class="op">)</span><span class="op">[</span></span>
<span class="r-in">  <span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/DelayedArray/man/DelayedMatrix-stats.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">mouseBrainSubsetSCE</span>, <span class="st">"counts"</span><span class="op">)</span><span class="op">)</span>,</span>
<span class="r-in">        decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span></span>
<span class="r-in"><span class="co">#subset to those first 100 genes</span></span>
<span class="r-in"><span class="va">subset</span> <span class="op">&lt;-</span> <span class="va">mouseBrainSubsetSCE</span><span class="op">[</span><span class="va">ord</span>, <span class="op">]</span></span>
<span class="r-in"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">scDiffExDESeq2</span><span class="op">(</span><span class="va">subset</span>, condition <span class="op">=</span> <span class="st">"level1class"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -- note: fitType='parametric', but the dispersion trend was not well captured by the</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    function: y = a/x + b, and a local regression fit was automatically substituted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    specify fitType='local' or 'mean' to avoid this message next time.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"mouseBrainSubsetSCE"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">scDiffExlimma</span><span class="op">(</span><span class="va">mouseBrainSubsetSCE</span>, condition <span class="op">=</span> <span class="st">"level1class"</span><span class="op">)</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Zero sample variances detected, have been offset away from zero</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"mouseBrainSubsetSCE"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">scDiffExANOVA</span><span class="op">(</span><span class="va">mouseBrainSubsetSCE</span>, condition <span class="op">=</span> <span class="st">"level1class"</span><span class="op">)</span></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

